src/api/feed-discovery/test/router.test.js

```diff
@@ -0,0 +1,163 @@
+        body: {
+          blogUrl: validUrl,
+        },
+      };
+      const mockRes = {};
+      const mockNextFunction = jest.fn();
+      checkValidUrl(mockReq, mockRes, mockNextFunction);
+      expect(mockNextFunction).toHaveBeenCalledTimes(1);
+      done();
+    });
+  });
+
+  describe('Test checkValidBlogPage middleware', () => {
+    it('should return 400 if the blog page responses with code other than 200', (done) => {
+      const InvalidBlogPage = 'https://notExistBlogPage.com/';
+
+      nock(InvalidBlogPage).get('/').reply(404);
+
+      request(app).post('/').send({ blogUrl: InvalidBlogPage }).expect(400, done);
+    });
+
+    it('should return 400 if the blog page responses with content type other than text/html', (done) => {
+      const InvalidBlogPage = 'https://notHtmlResponse.com/';
+
+      nock(InvalidBlogPage).get('/').reply(200, null, {
+        'Content-Type': 'text/xml',
+      });
+
+      request(app).post('/').send({ blogUrl: InvalidBlogPage }).expect(400, done);
+    });
+  });
+
+  describe('Test discoverFeedUrls middleware', () => {
+    it('should return 404 if there is no feed url discovered', (done) => {
+      const validUrl = 'https://LinkWithNoFeedUrls.com';
+      const mockBody = `
+      <html>
+        <head></head>
+        <body></body>
+      </html>
+    `;
+      const mockReq = {
+        body: {
+          blogUrl: validUrl,
+        },
+      };
+      const mockRes = {
+        locals: {
+          document: mockBody,
+        },
+      };
+      const mockNextFunction = jest.fn();
+      discoverFeedUrls(mockReq, mockRes, mockNextFunction);
+      const expectedError = createError(createError(404, 'No Feed Url Discovered'));
+      expect(mockNextFunction).toBeCalledWith(expectedError);
+      done();
+    });
+
+    it('should call next() if there is any discovered feed url', (done) => {
+      const validUrl = 'https://LinkWithFeedURLs.com';
+      const mockBody = `
+      <html>
+        <head>
+          <link rel="alternate" type="application/rss+xml" href="https://validBlog.com/feed"/>
+        </head>
+        <body></body>
+      </html>
+    `;
+      const mockReq = {
+        body: {
+          blogUrl: validUrl,
+        },
+      };
+      const mockRes = {
+        locals: {
+          document: mockBody,
+        },
+      };
+      const mockNextFunction = jest.fn();
+      discoverFeedUrls(mockReq, mockRes, mockNextFunction);
+      expect(mockNextFunction).toHaveBeenCalledTimes(1);
+      expect(mockRes.locals.feedUrls).toEqual(['https://validBlog.com/feed']);
+      done();
+    });
+  });
+
+  it('should return 200 status and if all middlewares got passed', async (done) => {
+    const blogUrl = 'https://test321.blogspot.com/';
+    const mockBlogUrlResponseBody = `
+      <!doctype html>
+      <html lang="en">
+        <head>
+          <link rel="alternate" type="application/atom+xml" href="https://test321.blogspot.com/feeds/posts/default/-/open-source"/>
```

Comment:
This is OK, but a tip: you're testing too many things at once.  A test should only test a single thing.  For example, you can test having no feed URLs (you have that), one RSS (you have that), one Atom, multiple feeds, tests for various types separated into different files.

If you do a giant test that fails, it's hard to debug: which part of the test is failing?  If you separate these out into smaller tests that only test 1 thing, it's way easier to figure out what's broken.